includes/self-update.php

```diff
@@ -0,0 +1,61 @@
+<?php
+add_filter( 'update_plugins_github.com', 'feedland_blogroll_self_update', 10, 4 );
+
+/**
+ * Check for updates to this plugin
+ *
+ * @param array  $update   Array of update data.
+ * @param array  $plugin_data Array of plugin data.
+ * @param string $plugin_file Path to plugin file.
+ * @param string $locales    Locale code.
+ *
+ * @return array|bool Array of update data or false if no update available.
+ */
```

Comment:
```suggestion
/**
 * Check for updates to this plugin
 *
 * @param array|false $update      Array of update data.
 * @param array       $plugin_data Array of plugin data.
 * @param string      $plugin_file Path to plugin file.
 * @param string[]    $locales     Locale code.
 *
 * @return array|false Array of update data or false if no update available.
 */
```

I'd update the doc block to match what's in the [docs](https://developer.wordpress.org/reference/hooks/update_plugins_hostname/).

It's nice to keep in mind that we're requiring PHP 7.4 but since it's EOL, we might want to consider requiring 8.1+ which would allow us to write the function using union types for type hints:

```php
function feedland_blogroll_self_update( array|false $update, array $plugin_data, string $plugin_file, array $locales ): array|false {
```

Reply:
> we might want to consider requiring 8.1+ which would allow us to write the function using union types for type hints:

This is a great idea. However, I recently got bitten by union return types on a plugin, since it got loaded on a site that we don't have control over, which is still on 7.4. Since this will be potentially widespread and public, I don't want to risk union type hints just yet. Maybe in a year :)